smart_open/tests/test_s3.py

```diff
@@ -320,6 +321,29 @@ def test_gzip(self):
 
         self.assertEqual(expected, actual)
 
+    def test_buffered_writer_wrapper_works(self):
+        """
+        Ensure that we can wrap a smart_open s3 stream in a subclass of
+        BufferedWriter. BufferedWriter passes a memoryview object to the
+        underlying stream in python >= 2.7
+        """
+
+        class BufferedWriterSubclass(io.BufferedWriter):
```

Comment:
Is there a reason why we need to subclass here? Why not just use io.BufferedWriter directly?

Reply:
Great point :)